### ISSUE TITLE
BadRequestException: Invalid model specified: Validation Result: warnings : [], errors : [Invalid model schema specified]

### ISSUE THREAD

**Issue by ibrt** [awslabs/aws-apigateway-swagger-importer#75]
I am trying to import the following API:

{
    "swagger": "2.0",
    "info": {
        "version": "0.0.1",
        "title": "Prototype Events API"
    },
    "paths": {
        "/events": {
            "get": {
                "description": "Gets all events in reverse chronological order.",
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "schema": {
                            "type": "array",
                            "items": {
                                "$ref": "#/definitions/Event"
                            }
                        }
                    },
                    "default": {
                        "description": "Unexpected error",
                        "schema": {
                            "$ref": "#/definitions/Error"
                        }
                    }
                }
            },
            "post": {
                "description": "Creates an event.",
                "parameters": [
                    {
                        "name": "Event",
                        "in": "body",
                        "description": "The event to be created.",
                        "required": true,
                        "schema": {
                            "$ref": "#/definitions/Event"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful response",
                        "schema": {
                            "$ref": "#/definitions/Event"
                        }
                    },
                    "default": {
                        "description": "Unexpected error",
                        "schema": {
                            "$ref": "#/definitions/Error"
                        }
                    }
                }
            }
        }
    },
    "definitions": {
        "Event": {
            "type": "object",
            "properties": {
                "userId": {
                    "type": "string",
                    "description": "Hardcoded to \"some-user\", for now do not use."
                },
                "eventId": {
                    "type": "string",
                    "description": "Id of the event."
                },
                "description": {
                    "type": "string",
                    "description": "Description of the event."
                },
                "imageUrl": {
                    "type": "string",
                    "description": "Url of the event image."
                },
                "when": {
                    "type": "number",
                    "description": "Event time as Unix timestamp."
                },
                "where": {
                    "$ref": "#/definitions/Location",
                    "description": "Event location."
                }
            }
         },
        "Location": {
            "type": "object",
            "properties": {
                "lat": {
                    "type": "number",
                    "description": "Latitude of the event location."
                },
                "lng": {
                    "type": "number",
                    "description": "Longitude of the event location."
                },
                "name": {
                    "type": "string",
                    "description": "Name of the event location."
                }
            }
        },
        "Error": {
            "type": "object",
            "properties": {
                "errorMessage": {
                    "type": "string",
                    "description": "Error message."
                },
                "errorType": {
                    "type": "string",
                    "description": "Error type."
                },
                "stackTrace": {
                    "type": "array",
                    "items": {
                        "type": "string"
                    },
                    "description": "Error stack trace."
                }
            }
        }
    }
}

It doesn't work. When I run the tool, I get this output:

2015-11-02 16:15:38,349 INFO - Attempting to create API from Swagger definition. Swagger file: ../eclipse/connect-lambda/swagger.yaml
reading from ../eclipse/connect-lambda/swagger.yaml
2015-11-02 16:15:38,436 INFO - Parsed Swagger with 1 paths
2015-11-02 16:15:38,439 INFO - Creating API with name Prototype Events API
2015-11-02 16:15:38,853 INFO - Removing default model Empty
2015-11-02 16:15:38,960 INFO - Removing default model Error
2015-11-02 16:15:39,065 INFO - Creating model for api id 3e1ov7sirk with name Event
2015-11-02 16:15:39,190 INFO - Generated json-schema for model Event: {"type":"object","properties":{"userId":{"type":"string","description":"Hardcoded to \"some-user\", for now do not use."},"eventId":{"type":"string","description":"Id of the event."},"description":{"type":"string","description":"Description of the event."},"imageUrl":{"type":"string","description":"Url of the event image."},"when":{"type":"number","description":"Event time as Unix timestamp."},"where":{"description":"Event location.","$ref":"#/definitions/Location"}},"definitions":{"Location":{"type":"object","properties":{"lat":{"type":"number","description":"Latitude of the event location."},"lng":{"type":"number","description":"Longitude of the event location."},"name":{"type":"string","description":"Name of the event location."}}}}}
2015-11-02 16:15:39,300 INFO - Creating model for api id 3e1ov7sirk with name Location
2015-11-02 16:15:39,306 INFO - Generated json-schema for model Location: {"type":"object","properties":{"lat":{"type":"number","description":"Latitude of the event location."},"lng":{"type":"number","description":"Longitude of the event location."},"name":{"type":"string","description":"Name of the event location."}},"definitions":{}}
2015-11-02 16:15:39,415 INFO - Creating model for api id 3e1ov7sirk with name Error
2015-11-02 16:15:39,420 INFO - Generated json-schema for model Error: {"type":"object","properties":{"errorMessage":{"type":"string","description":"Error message."},"errorType":{"type":"string","description":"Error type."},"stackTrace":{"type":"array","description":"Error stack trace.","items":{"type":"string"}}},"definitions":{}}
2015-11-02 16:15:39,522 ERROR - Error creating API, rolling back
com.amazonaws.services.apigateway.model.BadRequestException: Invalid model specified: Validation Result: warnings : [], errors : [Invalid model schema specified](Service: null; Status Code: 400; Error Code: null; Request ID: 036347dd-81c0-11e5-9c20-a7b0452cd4d8)
    at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1219)
    at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:803)
    at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:505)
    at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:317)
    at com.amazonaws.hal.client.HalClient.invoke(HalClient.java:235)
    at com.amazonaws.hal.client.HalClient.postResource(HalClient.java:98)
    at com.amazonaws.hal.client.HalResourceInvocationHandler.invoke(HalResourceInvocationHandler.java:117)
    at com.sun.proxy.$Proxy28.createModel(Unknown Source)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkApiImporter.createModel(ApiGatewaySdkApiImporter.java:139)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkSwaggerApiImporter.createModel(ApiGatewaySdkSwaggerApiImporter.java:108)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkSwaggerApiImporter.createModels(ApiGatewaySdkSwaggerApiImporter.java:101)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkSwaggerApiImporter.createApi(ApiGatewaySdkSwaggerApiImporter.java:65)
    at com.amazonaws.service.apigateway.importer.impl.ApiGatewaySwaggerFileImporter.importApi(ApiGatewaySwaggerFileImporter.java:48)
    at com.amazonaws.service.apigateway.importer.ApiImporterMain.execute(ApiImporterMain.java:137)
    at com.amazonaws.service.apigateway.importer.ApiImporterMain.main(ApiImporterMain.java:74)
2015-11-02 16:15:39,523 INFO - Deleting API 3e1ov7sirk
2015-11-02 16:15:39,995 ERROR - Error importing API definition
com.amazonaws.services.apigateway.model.BadRequestException: Invalid model specified: Validation Result: warnings : [], errors : [Invalid model schema specified](Service: null; Status Code: 400; Error Code: null; Request ID: 036347dd-81c0-11e5-9c20-a7b0452cd4d8)
    at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1219)
    at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:803)
    at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:505)
    at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:317)
    at com.amazonaws.hal.client.HalClient.invoke(HalClient.java:235)
    at com.amazonaws.hal.client.HalClient.postResource(HalClient.java:98)
    at com.amazonaws.hal.client.HalResourceInvocationHandler.invoke(HalResourceInvocationHandler.java:117)
    at com.sun.proxy.$Proxy28.createModel(Unknown Source)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkApiImporter.createModel(ApiGatewaySdkApiImporter.java:139)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkSwaggerApiImporter.createModel(ApiGatewaySdkSwaggerApiImporter.java:108)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkSwaggerApiImporter.createModels(ApiGatewaySdkSwaggerApiImporter.java:101)
    at com.amazonaws.service.apigateway.importer.impl.sdk.ApiGatewaySdkSwaggerApiImporter.createApi(ApiGatewaySdkSwaggerApiImporter.java:65)
    at com.amazonaws.service.apigateway.importer.impl.ApiGatewaySwaggerFileImporter.importApi(ApiGatewaySwaggerFileImporter.java:48)
    at com.amazonaws.service.apigateway.importer.ApiImporterMain.execute(ApiImporterMain.java:137)
    at com.amazonaws.service.apigateway.importer.ApiImporterMain.main(ApiImporterMain.java:74)

**Comment by rpgreen** [comment]
Curious, does it work if you take out "items" out of the Error model
properties?

**Comment by ibrt** [comment]
Actually it works if I take out "description" from Error/stackTrace. I moved it to Error/stackTrace/items and it works. I've tried the original in various JSON schema validators and they all say it's valid though...

**Comment by mrburrito** [comment]
I'm having the exact same problem. The interesting thing is, if I ran the importer a second time, it uploaded successfully. I could continually repeat that, run once and fail, run again and succeed. I finally figured out the `description` tag on an `array` property was causing the issue and moved my descriptions inside the `item` property. Just like @ibrt, when I pasted the JSONSchema generated from my Swagger file into a validator, it passed but failed when API Gateway attempted to import it.
